enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,525 @@
+    Reason:                UpdatePostActionCompleteReloadedCRIO
+    Status:                False
+    Type:                  ReloadedCRIO
+```
+
+The first option here indicates that this phase has never happened. The second one indicates that is has happend, just not during this update cycle. That is what the `Action during update to...` shows. That rendered config is not the one we are updating to currently.
+
+
+The MCO in 4.15 is aiming to use these objects to improve the source of truth for MCP reporting. If a user is opted into TechPreview, the MachineConfigPools will pull their `Updated`, `Updating`, and `Degraded` statuses from the MCN objects rather than from the nodes themselves. There are a few reasons for this: 
+
+1. Pulling the MCP state from the nodes in the pool results in reported behaviors that are outside of the MCOs control for example Cordoning and Draining. The MCO currently reports to Upgradeable=False in the CO when the node is Cordoned by an outside actor. This should only be the case if the MCO is attempting to Cordon and Drain a node. This also means that the Pool shows Updating=True which isn't the case.
+2. The MCO goes Upgradeable=False when new nodes are added. This is because these new nodes are reporting that they are not ready when they haven't even joined the Pool correctly yet. Since the MCP directly pulls its status from the nodes, this causes all sorts of issues. The MCN object will wait until the node is all settled in the pool before using it for state reporting meaning the MCO should show Upgradeable=True
+
+
+The MCN is meant to clarify state reporting for the users and disambiguate these edge cases.
+
+### API Extensions
+
+- Adds the MachineConfigNodes CRD
+
+### Risks and Mitigations
+
+There might be users who do not know to look here. We will mititagte this by reporting in the CO and/or the node status go to
+and look at this object for MCO related node failures or progression.
+
+Adding new MCD node states might have unintended consequences on the current flow which just has "Working" and "Done". Will make sure the
+states I add are either only in between Working and Done OR no matter what we end up back in the "Done" state.
+
+### Drawbacks
+
+The only drawback is increased API usage in the MCO. However, any approach to increased state reporting will increase API calls. This is the most well thought approach to this. 
+
+## Design Details
+
+### Open Questions [optional]
+
+None.
+
+### Test Plan
+
+MCO e2e tests and unit tests will cover this functionality.
+
+### Graduation Criteria
+
+This feature is behind the tech-preview FeatureGate in 4.15. Once it is tested by QE and users it can be GA'd since it should not impact daily usage of a cluster.
+
+## Dev Preview -> Tech Preview
+
+Not applicable. Feature introduced in Tech Preview. 
+
+## Tech Preview -> GA
```

Comment:
This might be a good place to track outstanding points to clarify, off the top of my head, I can think of:
* Clarifying the conditions and making sure they represent the current state, but with some history
* Making sure it's obvious what the current state of a resource is from the printer columms
  * Idle, Update Required, Updating <- I would want to know which of these states each MCN is in
  * Current and desired configs
  * High level state of update phases

Reply:
yeah. I think the main things are 

1. making sure the history is shown but the reason an message are clear in that this is a past event. And all of these events should be cleared when a new update begins
2. add an mcp column to track what pool the node is in?

Reply:
@JoelSpeed do you want me to represent these here? or can we keep track of this in Jira cards for future work?

Reply:
I think it would be good to add them as future work items here so we have it in writing that we've agreed on this being done